definition_validator/validator.cc

```diff
@@ -481,13 +485,19 @@ void validateSealed(core::Context ctx, const core::ClassOrModuleRef klass, const
     const auto superClass = klass.data(ctx)->superClass();
     if (superClass.exists() && superClass.data(ctx)->isClassOrModuleSealed()) {
         auto file = bestNonRBIFile(ctx, klass);
-        if (!absl::c_any_of(superClass.data(ctx)->sealedLocs(ctx), [file](auto loc) { return loc.file() == file; })) {
-            if (auto e =
-                    ctx.beginError(getAncestorLoc(ctx, classDef, superClass), core::errors::Resolver::SealedAncestor)) {
-                e.setHeader("`{}` is sealed and cannot be inherited by `{}`", superClass.show(ctx), klass.show(ctx));
-                for (auto loc : superClass.data(ctx)->sealedLocs(ctx)) {
-                    e.addErrorLine(loc, "`{}` was marked sealed and can only be inherited in this file",
-                                   superClass.show(ctx));
+        // HACK: Package.test.rbi may extend a class defined in Package.rbi because it contains a
+        // Package class only referenced by test exports.
```

Comment:
Is it not possible to guarantee that the sealed subclasses are defined in the same file as the parent class?

Reply:
`source.rb`:

```rb
class MyClass < SealedSubclass

end

class TestMyClass < SealedSubclass

end
```

`__package.rb`:

```
export MyClass
export_for_test TestMyClass
```

Now the RBI for the package contains `MyClass`, and the RBI for the test package contains `TestMyClass`, violating this invariant.

Reply:
Gotcha. The problem with this hack right now is that I don't think this should be an option for all people writing RBI files to circumvent `sealed!`'s invariants. If we could figure out concretely that an RBI file is a package-exported RBI file, that might be a better solution?